test-support/src/main/java/org/springframework/ldap/test/unboundid/EmbeddedLdapServer.java

```diff
@@ -102,4 +107,88 @@ public void shutdown() {
 		this.directoryServer.shutDown(true);
 	}
 
+	/**
+	 * Helper class for embedded Unboundid ldap server.
+	 *
+	 * @author Emanuel Trandafir
+	 * @since 3.3
+	 */
+	public static final class Builder {
+
+		private int port = 0;
+
+		private Consumer<InMemoryDirectoryServerConfig> configurationCustomizer = (__) -> {
+		};
+
+		private String partitionName = "jayway";
+
+		private final String partitionSuffix;
+
+		private Builder(String partitionSuffix) {
+			this.partitionSuffix = partitionSuffix;
+		}
+
+		public Builder port(int port) {
```

Comment:
Please add JavaDoc for each of the setter methods. In each JavaDoc, please make sure to indicate the default value.